freqtrade/freqtradebot.py

```diff
@@ -1358,7 +1358,6 @@ def _safe_exit_amount(self, pair: str, amount: float) -> float:
         trade_base_currency = self.exchange.get_pair_base_currency(pair)
         wallet_amount = self.wallets.get_free(trade_base_currency)
         logger.debug(f"{pair} - Wallet: {wallet_amount} - Trade-amount: {amount}")
-        # TODO-lev: Get wallet amount + value of positions
```

Comment:
I actually don't think we need to do anything here, the amount that's in positions shows up in `USED`, same as the amount which is in a limit order

```
DOT: { free: 0, used: 0, total: 0 },
  BTC: { free: 0, used: 0, total: 0 },
  SOL: { free: 0, used: 0, total: 0 },
  BNB: { free: 0, used: 0, total: 0 },
  ETH: { free: 0, used: 0, total: 0 },
  ADA: { free: 0, used: 0, total: 0 },
  USDT: { free: 58.54689751, used: 1082.52790274, total: 1113.9274002 },
  XRP: { free: 0, used: 0, total: 0 },
  BUSD: { free: 0.07715642, used: 0, total: 0.07715642 },
  timestamp: undefined,
  datetime: undefined,
  free: {
    DOT: 0,
    BTC: 0,
    SOL: 0,
    BNB: 0,
    ETH: 0,
    ADA: 0,
    USDT: 138.54689751,
    XRP: 0,
    BUSD: 0.07715642
  },
  used: {
    DOT: 0,
    BTC: 0,
    SOL: 0,
    BNB: 0,
    ETH: 0,
    ADA: 0,
    USDT: 1002.52790274,
    XRP: 0,
    BUSD: 0
  },
  total: {
    DOT: 0,
    BTC: 0,
    SOL: 0,
    BNB: 0,
    ETH: 0,
    ADA: 0,
    USDT: 1113.9274002,
    XRP: 0,
    BUSD: 0.07715642
  }
}
```

Reply:
that's wrong.

assuming you're in an XRP/USDT position - then the value we're looking for here is the XRP value - which is 0 in your case (there's no other value that would make sense either - so i'm taking XRP as sample).
it may show up as "used USDT" - but that's irrelevant - as we look for "currency" - not "amount.

It's currently not impacting because of the "trading_mode==Futures" below - but unless you can also remove the "or" condition below (which means something in the remaining code changed), this will have to stay as TODO-lev.

Reply:
I don't really understand what the problem is, it looks like this method checks for if you have enough currency to safely exit, but that's not an issue on futures, you can always close a position

Reply:
letme repeat what i said in another PR (again, it's REALLY tiring to explain the same thing to you over and over again!):
your assumption is that nothing else happens (like manual interference) - which we can't fully rely on.

What if the user closed the position manually? or it was liquidated? in both cases, the position would no longer be available, causing an exception (and potential other errors) while closing.

We're CURRENTLY excluding this check for futures - but that's a temporary measure until we use fetch_position to get the correct balance.

Reply:
The main point here is - without fetching positions - output of `/wallet` is wrong too (we discussed this last time as well).
once we have wallets uptodate - we can also check if we reallly still own the position.

Reply:
> it's REALLY tiring to explain the same thing to you over and over again!):

You have to be more polite. It's discouraging to hear that my work is a burden to you and it makes me want to stop contributing. I read all of your comments and think about them. When you didn't understand funding fees, I used every method I had to explain them to you.

--------

I do everything I can when reading Github comments, but if you want some tips on how to make explanations ultra clear, I recommend using more markdown syntax for things like

- bullet points
- **bolded words**
- sharing images

Reply:
I do believe that you read what i write - but it's either a understanding (didn't understand the content / intend) or a forgetting (forgot what you read a few minutes/hours/days later).


What i mean is this discussion: https://github.com/freqtrade/freqtrade/pull/6349#discussion_r798268236, which is a replica of the discussion above, just from 2ish weeks ago.
Using more markdown will not help it, as the content won't change (the discussion will still be the same). 

And this again proves my point - i feel like i'm repeating myself over the same exact topic quite often recently - often within the same PR. it's not the work that is a burden, but the "need to tell you x times" - which is simply an attention thing (it seems like you're flying over comments, and then forgetting half of it by the time you're done). 
i'm trying to tell you that it's tiring for reviewers when they have to repeat themselves again and again (either within a PR, or across multiple PR's ) on the same exact thing.


Once wallet includes positions, i don't see what would be wrong by including it here, too. 
Removing the TODO-lev feels like a "my goal is to get rid of the TODO-lev's, no matter what" - as it's in fact not done, and should stay there as reminder that we wanted to remove this.

I've tried the case where the position on exchange disappeared / was smaller - and received exceptions from the `create_order` method.
Freqtrade didn't crash, but it was also not properly handled, as we do know beforehand that the position is no longer open - at least once wallets output is correct.